psi4/src/psi4/libmints/mintshelper.cc

```diff
@@ -2337,6 +2337,106 @@ std::vector<SharedMatrix> MintsHelper::ao_potential_deriv1_helper(int atom) {
     return grad;
 }
 
+std::vector<SharedMatrix> MintsHelper::ao_overlap_half_deriv1_helper(const std::string &half_der_side, int atom) {
+    std::vector<std::string> cartcomp;
+    cartcomp.push_back("X");
+    cartcomp.push_back("Y");
+    cartcomp.push_back("Z");
+
+    std::shared_ptr<OneBodyAOInt> GInt(integral_->ao_overlap(1));
+
+    std::shared_ptr<BasisSet> bs1 = GInt->basis1();
+    std::shared_ptr<BasisSet> bs2 = GInt->basis2();
+
+    int nbf1 = bs1->nbf();
+    int nbf2 = bs2->nbf();
+
+    std::vector<SharedMatrix> grad;
+    for (int p = 0; p < 3; p++) {
+        std::stringstream sstream;
+        sstream << "ao_overlap_half_deriv1_" << atom << cartcomp[p];
+        grad.push_back(SharedMatrix(new Matrix(sstream.str(), nbf1, nbf2)));
```

Comment:
I think:
```
grad.push_back(std::make_shared<Matrix>(sstream.str(), nbf1, nbf2));
```
is the preferred idiom.

Reply:
Made this change at every instance I could find in mintshelper.cc